src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -0,0 +1,389 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="HttpClientRequestMessage.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Client.Serialization
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.OData.Client;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Threading.Tasks;
+    using Microsoft.OData;
+
+    /// <summary>
+    /// HttpClient based implementation of DataServiceClientRequestMessage.
+    /// </summary>
+    public class HttpClientRequestMessage : DataServiceClientRequestMessage, IDisposable

```

Comment:
I'm assuming by introducing the class `HttpClientRequestMessage`, the class, `HttpWebRequestMessage` should go. This will as well help to identify all other classes that still use `HttpWebRequest`.  There are [instances ](https://github.com/OData/odata.net/blob/6b4c4d9d8c249925de2814e7a15fb268b108c4e2/src/Microsoft.OData.Client/DataServiceContext.cs#L2779)in DataServiceContext class where `HttpWebRequest `is still being used.